python_translators/translators/best_effort_translator.py

```diff
@@ -1,85 +1,45 @@
-        self.translators.append(t)
+class BestEffortTranslator(CompositeParallelTranslator):
+    def __init__(self, source_language: str, target_language: str):
+        super(BestEffortTranslator, self).__init__(source_language=source_language, target_language=target_language)
 
-        # Add a microsoft translator
-        self.translators.append(MicrosoftTranslatorFactory.build(source_language, target_language))
-
-        # Add a microsoft translator that ignores context
-        t = MicrosoftTranslatorFactory.build(source_language, target_language)
-        t.add_query_processor(RemoveAllContext())
-        self.translators.append(t)
-
-        # Add a Glosbe translator
-        self.translators.append(GlosbeTranslator(source_language, target_language))
-
-    def _estimate_costs(self, query: TranslationQuery) -> TranslationCosts:
-        return TranslationCosts(
-            money=10
+        lang_config = dict(
+            source_language=source_language,
+            target_language=target_language
         )
 
-    def _translate(self, query: TranslationQuery) -> TranslationResponse:
-
-        translations = []
-        costs = TranslationCosts(money=0)  # no costs yet
-
-        max_translations = query.max_translations
-
-        for translator in self.translators:
-
-            # Reached the max number of translations?
-            if len(translations) >= max_translations:
-                break
-
-            query.max_translations = max_translations - len(translations)
-
-            # Perform translation
-            response = translator.translate(query)
-
-            assert len(response.translations) <= query.max_translations
-
-            # Add translations
-            for translation in response.translations:
-
-                if translation.lower() not in map(lambda s: s.lower(), translations):
-                    translations.append(translation)
-
-            # Add costs
-            costs.money += response.costs.money
-
-        return TranslationResponse(
-            translations=translations,
-            costs=costs
-        )
-
-
-if __name__ == '__main__':
-    t = BestEffortTranslator(source_language='nl', target_language='en')
-    response = t.translate(TranslationQuery(
-        before_context='De directeur',
-        query='treedt af',
-        after_context='',
-    ))
-
-    print(vars(response))
\ No newline at end of file
+        # Google Translator with context
+        t = GoogleTranslatorFactory.build_contextless(**lang_config)
+        t.quality = 80
+        self.add_translator(t)
+
+        # Google Translator without context
+        t = GoogleTranslatorFactory.build_with_context(**lang_config)
+        t.quality = 70
+        self.add_translator(t)
+
+        # Google Translator without context
+        t = MicrosoftTranslatorFactory.build_with_context(**lang_config)
+        t.quality = 60
+        self.add_translator(t)
+
+        # Google Translator without context
```

Comment:
It's a single line that's repeated. I don't see how I could get rid of repeating that line. Do you have any idea? :)